### ISSUE TITLE
add the function of check site symmetry

### ISSUE THREAD

**Issue by qzhu2017** [qzhu2017/crystallography#3]
Ideally, we need to check the site symmetry after we do each wyckoff merge.
No check is needed if there is no merge.

**Comment by qzhu2017** [comment]
@scottfredericks , I think you can enable the function of check_wyckoff_position() in the merge_coordinate function.
something like the followings,
```python
                merged = np.array(merged)
                if check_wyckoff_position(merged, sg, wyckoffs) is not False:
                    coor = merged
                else:
                    return coor, False
```

**Comment by qzhu2017** [comment]
@scottfredericks  please fix the following bug shown in bug.py

**Comment by qzhu2017** [comment]
in check_wyckoff_position()
```python
    for i, wp in enumerate(wyckoffs):
        w_symm = w_symm_all[i]
        if len(p_symm) == len(w_symm):
            temp = w_symm
            for p in p_symm:
                for w in temp:
                    if p == w:
                        temp.remove(w)
```
p/w contains the information for both rotation (rot) and translation (tau) operation.
sometimes, tau of p might return an array of [1.5, 1.5, 1.5]
while tau of w might be [0.5, 0.5, 0.5]
This should be equivalent. 
@scottfredericks please enable the periodic boundary condition check for this case.

**Comment by qzhu2017** [comment]
@scottfredericks 
Problem:
We generate site_symmetry from general wyckoff position, while we compare it with the data from a database in csv file.
However, they might differ in the translation vector.
This happens not only in A/B/C/F/I cells, also in P cells. Below is case for P4mm (99)
```
------ site symmetry from our code------------
['x, y, z', '-y+1, -x+1, z']
['x, y, z', '-y+1, -x+1, z']
['x, y, z', 'y, x, z']
['x, y, z', 'y, x, z']
------ site symmetry from database------------
['x, y, z', 'y, x, z']
['x, y, z', 'y, x, z']
['x, y, z', '-y, -x, z']
['x, y, z', '-y, -x, z']
```
Solution:
generate a new list of wyckoff database to make it is consistent with the way generated from general wyckoff position.

**Comment by scottfredericks** [comment]
It seems site_symm() is automatically converting negative coordinate values
into positive ones (due to SymmOp.from_xyz_string()). For example, [
0.23631801, -0.23631801, -0.06786002] is encoded as [ 0.23631801
0.76368199  0.93213998] within the function. As a result, the coordinates
are interpreted as 'x, -x+1, z' instead of 'x, -x, z'. I will see if there
is a way to prevent this automatic filtering.

**Comment by qzhu2017** [comment]
an attempt to fix the issue a99a03157e976840506335e1c41ae2821e272439

**Comment by scottfredericks** [comment]
The problem was actually in my code, not the SymmOp class. I removed a line
which filtered out negative values. The code now works for coordinates
generated directly from the Wyckoff position (using negative values for -x,
etc.) However, if we input only positive values, the code still produces
the same error. For now, we can avoid this by keeping negative values
during merge_coordinates. In the future, we need a way to recognize
already-filtered values; this will depend on the other coordinates being
checked, and will likely require trying a mapping for x, y, and z, and
making sure all other coordinates in the set work with the same mapping.

**Comment by scottfredericks** [comment]
I have updated wyckoff_list.csv to include values such as "x+1", and I've updated wyckoff_symmetry.csv based on this new wyckoff position databaes. As long as negative values are carried over from the general position, check_wyckoff_position now works. However, I am not sure if merge_coordinates uses this method, because check_wyckoff_position(merged, sg) still fails very often during crystal generation (note that the text output for the failure is currently commented out).

**Comment by qzhu2017** [comment]
@scottfredericks, I suggest add an option of considering periodic boundary condition here. When we check the site symmetry for the merged coordinate, we don't really need to know the exact wyckoff letters. So the ambiguity of two wyckoff sites with same rotational matrices and different translation vector is not an issue in this case.

```python
def check_wyckoff_position(points, sg, wyckoffs=None, pbc=False):
      """
      pbc: if we check the translation vector by periodic boundary condition. Default: False 
      """
```

**Comment by qzhu2017** [comment]
@scottfredericks  A recent test:

-----------------Species:[3], NumIons:[8]----------------------
Space group  requested:  2 generated 2 
Space group  requested:  3 generated 3 
Space group  requested:  4 generated 4 
Space group  requested:  5 generated 5 
something is wrong
[[ 0.85097362  0.43820877  0.86841429]
 [ 0.85097362 -0.43820877  0.86841429]]
**[[-0.14902638 -0.5        -0.13158571]]**
6

[x, 0, z] should be a valid wyckoff position for space group Pm

**Comment by qzhu2017** [comment]
@scottfredericks 
For space group 210, below is another set of data from my test,
[[-0.25 -0.25 -0.25]
 [ 0.25 -0.25  0.25]
 [ 0.25  0.25 -0.25]
 [-0.25  0.25  0.25]
 [-0.5   0.    0.  ]
 [ 0.5  -0.5   0.5 ]
 [ 0.    0.5   0.  ]
 [ 0.    0.   -0.5 ]]

This should be wyckoff 8b. But it was missed by your code.

**Comment by scottfredericks** [comment]
I have added another fix. The temporary arrays used within check_wyckoff_position were being generated using = instead of deepcopy. As a result, they were not being copied correctly. I have added a test script, test_wyckoff_check.py. This script generates points from Wyckoff positions using random (possibly negative) xyz coordinates, then randomly adds a value of (-1,0, or +1) to each point. Then, it runs check_wyckoff_position on these points. I have ran it twice, and it passed both times. It might be good to run it a few more times to make sure we aren't just lucky.
